ed25519_utils.go

```diff
@@ -0,0 +1,29 @@
+package jwt
+
+import (
+	"encoding/pem"
+	"golang.org/x/crypto/ed25519"
+)
+
+
+// Parse PEM encoded Elliptic Curve Private Key Structure
+func ParseED25519PrivateKeyFromPEM(key []byte) (*ed25519.PrivateKey, error) {
+	var block *pem.Block
+	if block, _ = pem.Decode(key); block == nil {
```

Comment:
This is not right. A PEM file merely encodes a binary string. OpenSSH produces a weird format:

```
"openssh-key-v1\0"
0x00000004 "none"
0x00000004 "none"
0x00000000
0x00000001
0x00000033 // length of next two values with their lengths
0x0000000b "ssh-ed25519"
0x00000020 [32 bytes] // the public key
0x00000040 [64 bytes] // the private key
0x00000016 [22 bytes] // some bytes of context, usually login@hostname plus padding
```

Now I wouldn't call that a normal PEM file. Usually, PEM files contain ASN.1-encoded data. Looking at a ECDSA key generated by ssh-keygen, I see it contains:

```
0x3081dc // 0xdc byte BMPString
0x0201 0x01 // Integer value 1
0x0442 [66 bytes] // private key
0xa007 [7 bytes] // something of type a0
0xa18189 // 137 bytes of type a1
0x038186 [134 bytes] // public key
```

Problem seems to be that EC keys have a format specified by RFC5915 and Ed keys don't. So OpenSSH made one up. Right now, that seems like the best we can do, so you'll need to be able to read OpenSSH keys.

Reply:
I generated the example one with OpenSSL if I remember correctly. I'm not all that deep into RFC standards. I barely scratched them enough to dabble this one together as you probably already inferred by my sloppy naming. I hope you can get some use out of the mess which this aparrently is.